### PR TITLE
Logging migration (force_log_printf)

### PR DESCRIPTION
**Pull request by why-keith**
Converts all instances of the `force_log_printf()` logging method with the newer `cemuLog_log()`.
The majority of the conversions were performed by a script, so any incorrect conversions are likely to show up repeatedly.

Part of #703

### CODE REVIEW

**Comment by why-keith** [comment]
Spotted an issue with the representation of hex values. Will fix

File: src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRendererCore.cpp [hunk 1]
```diff
@@ -677,7 +677,7 @@ VkDescriptorSetInfo* VulkanRenderer::draw_getOrCreateDescriptorSet(PipelineInfo*
 			info.imageView = nullTexture1D.view;
 			info.sampler = nullTexture1D.sampler;
 			textureArray.emplace_back(info);
-			// forceLog_printf("Vulkan-Info: Shader 0x%016llx uses 1D texture but bound texture has mismatching type (dim: 0x%02x)", shader->baseHash, textureView->gx2Dim);
+			// cemuLog_log(LogType::Force, "Vulkan-Info: Shader 0x{:016} uses 1D texture but bound texture has mismatching type (dim: 0x{:02x})", shader->baseHash, textureView->gx2Dim);
```
Comment (Crementif) [inline src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRendererCore.cpp:680]:
```suggestion
			// cemuLog_log(LogType::Force, "Vulkan-Info: Shader 0x{:016x} uses 1D texture but bound texture has mismatching type (dim: 0x{:02x})", shader->baseHash, textureView->gx2Dim);
```

File: src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.cpp [hunk 2]
```diff
@@ -2040,7 +2040,7 @@ void VulkanRenderer::PipelineCacheSaveThread(size_t cache_size)
 		}
 		catch (const std::exception& ex)
 		{
-			forceLog_printf("can't create vulkan pipeline cache directory \"%s\": %s", dir.generic_string().c_str(), ex.what());
+			cemuLog_log(LogType::Force, "can't create vulkan pipeline cache directory \"{}\": {}", dir.generic_string().c_str(), ex.what());
```
Comment (Crementif) [inline src/Cafe/HW/Latte/Renderer/Vulkan/VulkanRenderer.cpp:2043]:
```suggestion
			cemuLog_log(LogType::Force, "can't create vulkan pipeline cache directory \"{}\": {}", _pathToUtf8(dir), ex.what());
```

File: src/Cafe/OS/libs/coreinit/coreinit_FS.cpp [hunk 3]
```diff
@@ -186,7 +186,7 @@ namespace coreinit
 
 	void __FSErrorAndBlock(std::string_view msg)
 	{
-		forceLog_printf("Critical error in FS: %s", msg.data());
+		cemuLog_log(LogType::Force, "Critical error in FS: {}", msg.data());
```
Comment (Crementif) [inline src/Cafe/OS/libs/coreinit/coreinit_FS.cpp:189]:
```suggestion
		cemuLog_log(LogType::Force, "Critical error in FS: {}", msg);
```

File: src/Cafe/OS/libs/coreinit/coreinit_FS.cpp [hunk 4]
```diff
@@ -626,15 +626,16 @@ namespace coreinit
...
-			forceLog_printf("Printing CMD block: ");
+			cemuLog_log(LogType::Force, "Corrupted FS command detected in stage {}", stage);
+			cemuLog_log(LogType::Force, "Printing CMD block: ");
 			for (uint32 i = 0; i < (sizeof(FSCmdBlockBody_t) + 31) / 32; i++)
 			{
 				uint8* p = ((uint8*)fsCmdBlockBody) + i * 32;
-				forceLog_printf("%04x: %02x %02x %02x %02x - %02x %02x %02x %02x - %02x %02x %02x %02x - %02x %02x %02x %02x | %02x %02x %02x %02x - %02x %02x %02x %02x - %02x %02x %02x %02x - %02x %02x %02x %02x",
-					i * 32,
-					p[0], p[1], p[2], p[3], p[4], p[5], p[6], p[7], p[8], p[9], p[10], p[11], p[12], p[13], p[14], p[15],
+				cemuLog_log(LogType::Force, "{:04}: {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02} | {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02} - {:02} {:02} {:02} {:02}",
```
Comment (Crementif) [inline src/Cafe/OS/libs/coreinit/coreinit_FS.cpp:634]:
```suggestion
				cemuLog_log(LogType::Force, "{:04x}: {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x} | {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x} - {:02x} {:02x} {:02x} {:02x}", 
```

File: src/audio/XAudio2API.cpp [hunk 5]
```diff
@@ -258,7 +258,7 @@ const std::vector<XAudio2API::DeviceDescriptionPtr>& XAudio2API::RefreshDevices(
 	}
 	catch (const std::system_error& ex)
 	{
-		forceLog_printf("XAudio2API::RefreshDevices: error while refreshing device list (%s - code: 0x%08x)", ex.what(), ex.code().value());
+		cemuLog_log(LogType::Force, "XAudio2API::RefreshDevices: error while refreshing device list ({} - code: 0x{:08})", ex.what(), ex.code().value());
```
Comment (Crementif) [inline src/audio/XAudio2API.cpp:261]:
```suggestion
		cemuLog_log(LogType::Force, "XAudio2API::RefreshDevices: error while refreshing device list ({} - code: 0x{:08x})", ex.what(), ex.code().value());
```

File: src/gui/GameUpdateWindow.cpp [hunk 6]
```diff
@@ -105,7 +105,7 @@ bool GameUpdateWindow::ParseUpdate(const fs::path& metaPath)
 		}
 		catch (const std::exception& ex)
 		{
-			forceLog_printf("GameUpdateWindow::ParseUpdate exist-error: %s at %s", ex.what(), target_location.generic_u8string().c_str());
+			cemuLog_log(LogType::Force, "GameUpdateWindow::ParseUpdate exist-error: {} at {}", ex.what(), target_location.generic_string());
```
Comment (Crementif) [inline src/gui/GameUpdateWindow.cpp:108]:
```suggestion
			cemuLog_log(LogType::Force, "GameUpdateWindow::ParseUpdate exist-error: {} at {}", ex.what(), _pathToUtf8(target_location));
```

File: src/gui/TitleManager.cpp [hunk 7]
```diff
@@ -535,7 +535,7 @@ void TitleManager::OnSaveDelete(wxCommandEvent& event)
 	}
 
 	if (!meta_file_edited)
-		forceLog_printf("TitleManager::OnSaveDelete: couldn't delete save entry in saveinfo.xml: %s", saveinfo.generic_u8string().c_str());
+		cemuLog_log(LogType::Force, "TitleManager::OnSaveDelete: couldn't delete save entry in saveinfo.xml: {}", saveinfo.generic_string());
```
Comment (Crementif) [inline src/gui/TitleManager.cpp:538]:
```suggestion
		cemuLog_log(LogType::Force, "TitleManager::OnSaveDelete: couldn't delete save entry in saveinfo.xml: {}", _pathToUtf8(saveinfo));
```

File: src/gui/dialogs/SaveImport/SaveImportWindow.cpp [hunk 8]
```diff
@@ -307,7 +307,7 @@ void SaveImportWindow::OnImport(wxCommandEvent& event)
 					timestamp.text().set(fmt::format("{:016x}", coreinit::coreinit_getOSTime() / ESPRESSO_TIMER_CLOCK).c_str()); // TODO time not initialized yet?
 					
 					if(!doc.save_file(saveinfo.c_str()))
-						forceLog_printf("couldn't insert save entry in saveinfo.xml: %s", saveinfo.generic_u8string().c_str());
+						cemuLog_log(LogType::Force, "couldn't insert save entry in saveinfo.xml: {}", saveinfo.generic_string());
```
Comment (Crementif) [inline src/gui/dialogs/SaveImport/SaveImportWindow.cpp:310]:
```suggestion
						cemuLog_log(LogType::Force, "couldn't insert save entry in saveinfo.xml: {}", _pathToUtf8(saveinfo));
```

File: src/gui/dialogs/SaveImport/SaveTransfer.cpp [hunk 9]
```diff
@@ -181,7 +181,7 @@ void SaveTransfer::OnTransfer(wxCommandEvent& event)
 	}
 
 	if (!meta_file_edited)
-		forceLog_printf("SaveTransfer::OnTransfer: couldn't update save entry in saveinfo.xml: %s", saveinfo.generic_u8string().c_str());
+		cemuLog_log(LogType::Force, "SaveTransfer::OnTransfer: couldn't update save entry in saveinfo.xml: {}", saveinfo.generic_string());
```
Comment (Crementif) [inline src/gui/dialogs/SaveImport/SaveTransfer.cpp:184]:
```suggestion
		cemuLog_log(LogType::Force, "SaveTransfer::OnTransfer: couldn't update save entry in saveinfo.xml: {}", _pathToUtf8(saveinfo));
```

**Review by Crementif — CHANGES_REQUESTED** [review]
Sorry for the late review! Here's some small changes but other then that it looks good!
 * I removed some c_str() since fmt has native C++ string support.
 * I also found some fs::path -> string conversions.

**Comment by why-keith** [comment]
> Sorry for the late review! Here's some small changes but other then that it looks good!

No worries, bit of mammoth to review this one. By the looks of things the other log types won't be half as bad.
I'll get to work on your suggestions.

File: src/config/LaunchSettings.cpp [hunk 10]
```diff
@@ -253,7 +253,6 @@ bool LaunchSettings::ExtractorTool(std::wstring_view wud_path, std::string_view
 		catch (const std::exception& ex)
 		{
 			cemuLog_log(LogType::Force, "can't write file: {}", ex.what());
-			puts(fmt::format("can't write file: %s\n", ex.what()).c_str());
```
Comment (Crementif) [inline src/config/LaunchSettings.cpp:256]: Are you sure this line isn't used? If I had to guess puts might be used here to output things to the command line. I'd probably just leave it in.

Comment (why-keith) [reply src/config/LaunchSettings.cpp:256]: I wasn't sure but it showed up in your recommended changes so I assumed that it wasn't required.
I'll stick it back in.

**Review by Crementif — APPROVED** [review]
Looks good to me!

**Pull request merged**